### ISSUE TITLE
TreeMap with children export to CSV is broken

### ISSUE THREAD

**Issue by fabiosangregorio** [amcharts/amcharts4#1312]
Exporting a TreeMap chart with children to CSV using the ExportMenu results in the following CSV:

```CSV
name,children
Core,"[object Object],[object Object],[object Object],[object Object],[object Object]"
```

I would expect a more usereful CSV export, even if I understand is not the most suitable format with nested data.

Live example: https://codepen.io/fabiosangregorio/pen/vwxVYg

**Comment by martynasma** [comment]
Indeed.

Although, I'm not entirely sure how to export hierarchical data to CSV.

I suppose charts like that should have CSV export option disabled.

Do you agree?

**Comment by martynasma** [comment]
I suppose this goes for Excel export, too.

**Comment by martynasma** [comment]
Sorry, for the flood. Now that I'm thinking about it, outright disabling CSV/XSLX export is not right. For a single-level data those would be legit exports.

This calls for some fuzzy logic. We'll have to give it some thinking :)

Will keep  you posted.

**Comment by fabiosangregorio** [comment]
Yes, Excel export doesn't work as well.

Can this type of chart (with hierarchical data) be created loading data from CSV? [The article](https://www.amcharts.com/docs/v4/concepts/data/loading-external-data/#CSV) on the docs regarding CSV data doesn't specify it.

If you can load hierarchical data from CSV I think you should also be able to export it, otherwise I agree with you.

Anyway, I am not familiar with the internal logic of amchart's data loading, but one structure I think would be feasible is to have parent/children relationships like this:

Tree:
```
         A
       /   \
      B     C
      |   / | \
      D  E  F  G
```
CSV:
```
id, name, parent
D, Daniel, B
B, Barbara, A
E, Eleonor, C
F, Fabio, C
G, Gabriel, C
C, Carl, A
A, Alan,
```

**Comment by martynasma** [comment]
Answering your question: no, you can't load hierarchical data from CSV.

I suppose the most logical export strategy for multi-level data to CSV is exporting only top-level, omitting fields that contain sub-data (children).

**Comment by fabiosangregorio** [comment]
Yes, I agree. Thank you for your time.

**Comment by martynasma** [comment]
Fixed in [4.4.7](https://github.com/amcharts/amcharts4/releases/tag/4.4.7).

## [4.4.7] - 2019-05-21

### Changed
- Export: Children columns on hierarchical charts (`TreeMap`, `Sunburst`, `ForceDirectedTree`) are no longer exported to "flat" formats (CSV, XLSX).

### Fixed
- `keepTargetHover` was hiding tooltip when transiting hover from one `Sprite` to another.
- Firefox was not displaying texts along paths on a page with `base href` set, due to Firefox [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=455986).
- `SunburstSeries` with 0 values were producing a JS error.
- Width (or height if vertical) of `Scrollbar`/`Slider` was always 100%, even if it was set to less than 100%.
- `WordCloud` with long labels that did not fit into container could get into infinite loop and never get drawn.
- If a lot of `GaugeChart` instances with axis bands were rendered on a single page, stack owerflow error could happen.
- Adding data to a chart with `XYChartScrollbar` and series in the scrollbar forced full-redraw of scrollbar chart which was making it blink.
- Z-indexing of `ColumnSeries3D` columns was incorrect in some cases.
- Tooltip of a 3D slice was not in the center of a slice.
- Changing of Series' name was not propagating to legend.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by OGudkov-TechHub** [comment]
Hi,

I found this issue during searching an answer for another question, but would like to share my recent experience with the topic. Maybe it will be helpful to someone.

In our project we are using Sunburst diagram and provide a possibility to do export of the data. The CSV/XLSX export produces "meaningless" (sorry) tables in result. I've resolved it with the custom chart exporting adapter (amCharts 4): I do prepare an outcome structure myself.

Sunburst data, in our case, have got the following structure:
```
    [{
      name: "Third",
      children: [
        {
          name: "C1",
          children: [
            { name: "EE1", value: 130 },
            { name: "EE2", value: 87 },
            { name: "EE3", value: 55 }
          ]
        },
        { name: "C2", value: 148 },
        {
          name: "C3", children: [
            { name: "CC1", value: 53 },
            { name: "CC2", value: 30 }
          ]
        },
        { name: "C4", value: 26 }
      ]
    }, ...
    ]
```
The outcome structure designed as follows:


Level 1 | Level 2 | Level 3 | Value
-- | -- | -- | --
Third | C1 | EE1 | 130
Third | C1 | EE2 | 87
Third | C1 | EE3 | 55
Third | C2 |   | 148
Third | C3 | CC1 | 53
Third | C3 | CC2 | 30
Third | C4 |   | 26

Table headers uses Sunburst's levels as labels. In case of XLSX export - this structure can be easily grouped by the Excel grouping feature.

Btw, the Heatmap chart export also produces not really friendly result. Resolved it in the same manner via custom adapter.
